Code/Framework/AzCore/AzCore/Debug/TraceMessagesDriller.cpp

```diff
@@ -9,94 +9,91 @@
 #include <AzCore/Debug/TraceMessagesDriller.h>
 #include <AzCore/Math/Crc.h>
 
-namespace AZ
+namespace AZ::Debug
 {
-    namespace Debug
+    //=========================================================================
+    // Start
+    // [2/6/2013]
+    //=========================================================================
+    void TraceMessagesDriller::Start(const Param* params, int numParams)
     {
-        //=========================================================================
-        // Start
-        // [2/6/2013]
-        //=========================================================================
-        void TraceMessagesDriller::Start(const Param* params, int numParams)
-        {
-            (void)params;
-            (void)numParams;
-            BusConnect();
-        }
+        (void)params;
+        (void)numParams;
+        BusConnect();
+    }
 
-        //=========================================================================
-        // Stop
-        // [2/6/2013]
-        //=========================================================================
-        void TraceMessagesDriller::Stop()
-        {
-            BusDisconnect();
-        }
+    //=========================================================================
+    // Stop
+    // [2/6/2013]
+    //=========================================================================
+    void TraceMessagesDriller::Stop()
+    {
+        BusDisconnect();
+    }
 
-        //=========================================================================
-        // OnAssert
-        // [2/6/2013]
-        //=========================================================================
-        void TraceMessagesDriller::OnAssert(const char* message)
-        {
-            // Not sure if we can really capture assert since the code will stop executing very soon.
-            m_output->BeginTag(AZ_CRC("TraceMessagesDriller", 0xa61d1b00));
-            m_output->Write(AZ_CRC("OnAssert", 0xb74db4ce), message);
-            m_output->EndTag(AZ_CRC("TraceMessagesDriller", 0xa61d1b00));
-        }
+    //=========================================================================
+    // OnAssert
+    // [2/6/2013]
+    //=========================================================================
+    void TraceMessagesDriller::OnAssert(const char* message)
+    {
+        // Not sure if we can really capture assert since the code will stop executing very soon.
+        m_output->BeginTag(AZ_CRC_CE("TraceMessagesDriller"));
```

Comment:
`AZ_CRC_CE` is a bit more expensive compilation-wise (its a compile-time crc whereas `AZ_CRC` is pre-computed)
It also has the issue that the CRC is not "searchable" (some logs will print the CRC and you would have to guess where is coming from).
Because of the above reasons, is somehow less preferred...
Dont worry about these ones because the drillers are going to be removed. Just wanted to let you know of the trade-offs

Reply:
I have metrics on `AZ_CRC_CE`
The additional compile time was negligible for using `AZ_CRC_CE`
[Crc32CompileTimings.xlsx](https://github.com/o3de/o3de/files/7521849/Crc32CompileTimings.xlsx)
e

Reply:
Thanks for bringing that up, I had wrong assumptions.
For the searchable part, I am referring to `AZ_CRC` when the CRC is actually passed. We can add a python script to generate them and leave it out of the build process.
I know the lack of that CRC in the code is a concern for others, and I agree we can solve it other ways.
Dropping my observations.

Reply:
We can do the same for `AZ_CRC_CE` as well, which is to run a pass over it to generate the Crc32 values

We could alos integrate a step in CMake configuration where it generates generate a .txt with just a list of calculated Crc32 values

Reply:
the generation of the crc values into a text file could be the solution for the reverse-lookup problem, really like that
Doesnt need to be a cmake target, we can just have a script that generates it, developers can run it when they need it.

Reply:
We might consider extending CRC support to:
- add `_crc` string suffix ( `"some_string"_crc` ) that resolves to a constexpr Crc.
- optional reverse lookup ( crc->string ) functionality ( either static/constexpr crc->string map, or runtime )

Runtime support would cost us some performance on Crc object construction, but IMHO we should care about comparisons/hashes of Crc objects. Supporting runtime reverse mapping will be needed anyway if we want dll-only gem distributions.

In addition, I think we have 2 orthogonal things going on in our Crc32 class:
- constexpr hashed string 
- crc32 utility class ( calculating crc32 for arbitrary data )

We should consider splitting this two into separate things.

Reply:
We actually have a constexpr hash string method defined in [AZStd::string_view](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/std/string/string_view.h#L871-L897).

You are correct that the Crc32 class should just be used for calculating checksums of blobs of data instead of being used as a way to hash strings.
The [AZ::Name](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Name/Name.h#L24-L33) class and the NameDictionary is better at storing a mapping of strings to hash values